cromwell_tools/tests/test_cromwell_auth.py

```diff
@@ -0,0 +1,73 @@
+import tempfile
+import json
+
+import pytest
+
+import requests_mock
+
+from .._cromwell_auth import CromwellAuth
+
+
+def setup_auth_types():
+    temp_dir = tempfile.mkdtemp()
+    secrets_file = temp_dir + 'fake_secrets.json'
+    username = "fake_user"
+    password = "fake_password"
+    url = "https://fake_url"
+
+    user_password = {
+        "url": url,
+        "user": username,
+        "password": password
+    }
+    with open(secrets_file, 'w') as f:
+        json.dump(user_password, f)
+
+    # produce authentication types
+    return {
+        "secrets_file": {"secrets_file": secrets_file},
+        "caas_key": {"caas_key": "path/fake_key.json", "cromwell_url": url},
+        "user_password": user_password
+    }
+
+auth_types = setup_auth_types()
+
+
+@pytest.fixture(scope='module', params=auth_types.values(), ids=auth_types.keys())
+def auth_types(request):
+    return request.param
+
+
+def test_cromwell_auth(auth_types):
+
+    def _request_callback(request, context):
+        context.status_code = 200
+        context.headers['test'] = 'header'
+        return {'request': {'body': "content"}}
+
+    def _request_callback_status(request, context):
+        context.status_code = 200
+        context.headers['test'] = 'header'
+        return {'status': 'Succeeded'}
+
+    with requests_mock.mock() as mock_request:
+        CromwellAuth(**auth_types)
+
+
+@requests_mock.mock()
+def test_get_workflow_statuses(self, mock_request):
```

Comment:
This seems like it would make more sense to put this in `test_cromwell_tools.py` instead.